src/pipdeptree/_models/dag.py

```diff
@@ -110,16 +107,10 @@ def filter_nodes(self, include: list[str] | None, exclude: set[str] | None) -> P
         if include is None and exclude is None:
             return self
 
-        # Note: In following comparisons, we use lower cased values so
-        # that user may specify `key` or `project_name`. As per the
-        # documentation, `key` is simply
-        # `project_name.lower()`. Refer:
-        # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#distribution-objects
         include_with_casing_preserved: list[str] = []
-        if include:
-            include_with_casing_preserved = include
-            include = [s.lower() for s in include]
-        exclude = {s.lower() for s in exclude} if exclude else set()
+        include_with_casing_preserved = include
+        exclude = {pep503_normalize(s) for s in exclude} if exclude else set()
+        include = [pep503_normalize(i) for i in include] if include else list()
```

Comment:
As I said it would be best to not touch on areas that aren't related to the PR like this one.

But just to mention a few problems, it seems you assign `include_with_casing_preserved` to an empty list then to `include` immediately afterwards. We could also crash in this function now because when we try to iterate over `include_with_casing_preserved` in the end of the function, `include` could be `None` (check out the type of `include`, it could be either a `list` or `None`).

Reply:
I revert my change, and only leave `pep503_normalize` part.